### DISCUSSION TITLE
Is it possible to make a form field disabled only on edit but not on create?

### DISCUSSION THREAD

**Discussion by mattsims** [filamentphp/filament#2041] [Help] [answered]
I'd like users to have to set a relation field when creating a new model, but then not have the ability to change the relation later.

I know the `->disabled()` option can take a boolean to enable/disable it dynamically, so is there a way to know inside the `form()` method whether we're creating a new resource or editing an existing resource?

Many thanks.

**Comment by danharrin** [comment, marked as answer]
Hey! `->disabled(fn ($livewire) => $livewire instanceof EditRecord)`

https://filamentphp.com/docs/2.x/forms/advanced#using-closure-customisation

**Reply by mattsims** [reply]
Fantastic, thank you! 🙏
